src/dodal/devices/i22/fswitch.py

```diff
@@ -0,0 +1,43 @@
+import asyncio
+import time
+from enum import Enum
+from typing import Dict
+
+from bluesky.protocols import Reading
+from ophyd_async.core import StandardReadable
+from ophyd_async.core.device import DeviceVector
+from ophyd_async.epics.signal import epics_signal_r
+
+
+class FilterState(str, Enum):
+    IN = "IN"
+    OUT = "OUT"
+
+
+class FSwitch(StandardReadable):
+    """
+    Device for i22's fswitch. This currently only implements the minimum
+    functionality for retrieving the number of lenses inserted.
+
+    Eventually this should be combined with the transfocator device in the i04
+    module but is currently incompatible as the Epics interfaces are different.
+    """
+
+    NUM_FILTERS = 128
+
+    def __init__(self, prefix: str, name: str = "") -> None:
```

Comment:
Nit: I'd prefer for top-level devices like this to require a name, but it looks like the Slits made it in with a default so I can't point to precedence =(

Reply:
@coretl thinks the opposite so we should probably resolve this at some point. Large point of diverging effort otherwise.

Reply:
Does this need to be recorded in a ticket or is it already noted somewhere?

Reply:
In https://github.com/DiamondLightSource/dodal/issues/415#issuecomment-2043044664 I suggest using the name of the device factory function to name the device. This would mean you could do:

```python
# i22.py
@device_factory(lazy=True)
def fswitch():
    return FSwitch(f"{BL}-MO-FSWT-01:")

assert fswitch().name == "fswitch"
```

If @DominicOram is happy with this approach I will write this up as an ADR for Dodal along with the connection strategy

Reply:
Yes, I believe that's the default we've tried to go with. Thank you @coretl